cmd/infra/powervs/create.go

```diff
@@ -1554,10 +1560,12 @@ func (infra *Infra) createTransitGateway(ctx context.Context, options *CreateInf
 	}
 
 	log(options.InfraID).Info("Creating Transit Gateway ...")
+	// Checking if global routing required for transit gateway.
+	globalRouting := regionutils.IsGlobalRoutingRequiredForTG(options.Region, options.VPCRegion)
 	tg, _, err = tgapisv1.CreateTransitGatewayWithContext(ctx, &transitgatewayapisv1.CreateTransitGatewayOptions{
 		Location:      utilpointer.String(options.TransitGatewayLocation),
 		Name:          utilpointer.String(transitGatewayName),
-		Global:        utilpointer.Bool(true),
+		Global:        utilpointer.Bool(globalRouting || options.TransitGatewayGlobalRouting),
```

Comment:
Won't it return false even when it actually requires True, in case of True returned from region map and false from options?

Reply:
It will return true right as its OR. Should it be the case that global-routing-options value overrides the one received from ```IsGlobalRoutingRequiredForTG``` api ?

Reply:
Oh sry, my bad. Yeah we need to override the options value when it's set to true. I think it will do that.